### PR TITLE
Use 'golden' files for amd and cjs test reference modules

### PR DESCRIPTION
**Pull request by johnjbarton**

### CODE REVIEW

**Comment by johnjbarton** [comment]
Pls review this one first then I will rebase #1652

File: test/unit/node/generated-code-dependencies.js [hunk 1]
```diff
@@ -347,17 +347,26 @@ suite('context test', function() {
     });
   });
 
+  function checkFile(outDir, goldenDir, filename) {
+    var actualContents = fs.readFileSync(path.resolve(outDir, filename));
+    var actual = actualContents.toString().split('\n');
```
Comment (arv) [inline test/unit/node/generated-code-dependencies.js]: For these I think splitting is going to lead to worse error message. The standard one shows the diff and by splitting we miss cross line diffs.

**Comment by arv** [comment]
LGTM. Much nicer.

File: test/unit/node/generated-code-dependencies.js (hunk 1 above)
Comment (johnjbarton) [inline test/unit/node/generated-code-dependencies.js]: I added a single character to on of the golden files. With the current splitting:
```
  1) context test compile module dir option AMD:

      Uncaught AssertionError: expected '  X"use strict";' to equal '  "use strict";'
      + expected - actual

      +  "use strict";
      -  X"use strict";
```
With `assert.deepEqual(actualContents.toString(), expectedContents.toString());`:
```
1) context test compile module dir option AMD:

      Uncaught AssertionError: expected 'define(["./dep"], function($__0) {\n  "use strict";\n  if (!$__0 || !$__0.__esModule)\n    $__0 = {default: $__0};\n  var q = $__0.q;\n  var p = \'module\';\n  return {\n    get p() {\n      return p;\n    },\n    __esModule: true\n  };\n});\n' to deeply equal 'define(["./dep"], function($__0) {\n  X"use strict";\n  if (!$__0 || !$__0.__esModule)\n    $__0 = {default: $__0};\n  var q = $__0.q;\n  var p = \'module\';\n  return {\n    get p() {\n      return p;\n    },\n    __esModule: true\n  };\n});\n'
      + expected - actual

      +"define([\"./dep\"], function($__0) {\n  X\"use strict\";\n  if (!$__0 || !$__0.__esModule)\n    $__0 = {default: $__0};\n  var q = $__0.q;\n  var p = 'module';\n  return {\n    get p() {\n      return p;\n    },\n    __esModule: true\n  };\n});\n"
      -"define([\"./dep\"], function($__0) {\n  \"use strict\";\n  if (!$__0 || !$__0.__esModule)\n    $__0 = {default: $__0};\n  var q = $__0.q;\n  var p = 'module';\n  return {\n    get p() {\n      return p;\n    },\n    __esModule: true\n  };\n});\n"
```
The split seems better in this comparison. Did you have something else in mind?

Comment (arv) [inline test/unit/node/generated-code-dependencies.js]: Hm, I was sure I had seen mocha use string diffs for the error message. I guess that might only be for some reporters.

Keep the split then.

**Pull request closed without merging**